ocaml/runtime/compact.c

```diff
@@ -261,16 +261,18 @@ static void do_compaction (intnat new_allocation_policy)
 
           if (t == Closure_tag){
             /* Revert the infix pointers to this block. */
-            mlsize_t i, startenv;
+            mlsize_t i;
             value v;
 
             v = Val_hp (p);
-            startenv = Start_env_closinfo (Closinfo_val (v));
             i = 0;
             while (1){
-              int arity = Arity_closinfo (Field (v, i+1));
+              int arity;
+              uintnat closinfo = Field (v, i+1);
+              if (Is_last_closinfo (closinfo)) break;
+              arity = Arity_closinfo (closinfo);
```

Comment:
Maybe I'm misreading, but this doesn't look right. The old code stops when `i >= startenv`, so when it has advanced past the infix closures, but the new code stops as soon as `Is_last_closinfo` holds. It looks like the new code doesn't actually process the last closure?

Is this codepath tested?

Reply:
I did in fact misread this, it's fine.